Documentation/Signoffs/Image_Processing.md

```diff
@@ -0,0 +1,113 @@
+
+1. In order for the system to properly detect the golf ball and extract the necessary information for aiming, the system needs to distinguish the golf ball
+   from the rest of the image.
+2. The program needs to use the coordinates from the camera to provide coordinates for aiming at the fishing line the golf ball is on which are about four
+   inches apart at the start of the trajectory and know which variable height the ball is out which different by about seven inches. If the coordinates are
+   off by an inch, the wire and height can be assumed to get the position of the golf ball.
+3. The fastest speed of the golf ball is 1.95 seconds from empirical data from the customer. The ball needs to be detected in enough time for the team to
+   aim the launcher and launch the projectile. This minimum allows time for the motors to make adjustments and fire after the information has been recieved
+   and interpreted which is expected to take half a second each.
+
+**Buildable Schematic**
+
+The system is purely software. Any connections for the processor should be shown in that subsystems schematic.
+
+A flow chart of the code is given below.
+
+![Function](../Images/Image_Processing/Flow_Chart.PNG)
+
+This shows the steps for each major calculation and what is necessary for each calculation. This process will be done twice to recieve two positions for the
+speed calculation done by the processor.
+
+**Analysis:**
+
+*Detection*
+
+The golf ball will be wrapped in foil and painted matte red. This will allow the object to be tracked using color. This can be done using a red, green, and
+blue color detection. The boundaries for the color are determined and entered in as limits. For these purposes, red will have the highest boundaries while
+green and blue will be much lower but still there to account for slight shading differences in the environment. These parameters and the image are then sent
+to a function called inRange. This will apply a mask to the array of pixels repersenting the image with a maske to determine the objects with the color.
+The results of such operations and algorithms can be seen below.
+
+![Function](../Images/Image_Processing/Detecting_Red.png)
+
+This can be adjusted to detect the ball at longer distances when the amount of pixels repersenting the object are smaller. The Big O analysis of this
+function would be O(n) because it is an iterative function that will go through all of the pixels provided [1]. For more cluttered environments with
+variety of color and lighting, objects can still be detected and tracked. An example of this can be seen below [2].
+
+![Function](../Images/Image_Processing/Detecting_Clutter.png)
+
+This uses the same algorithm, but applies it to an environment with varying light sources and colored backgrounds showing that the ball can still be
+identified.
+
+
+*Distance and Coordinates*
```

Comment:
This does not make sense. You are referring to the the width of a pixel... What does that mean? A pixel has no defined width.